test/sym/assume.wast

```diff
@@ -0,0 +1,17 @@
+(module
+  (import "symbolic" "i32_symbol" (func $i32_symbol (result i32)))
+  (import "symbolic" "assume" (func $assume_i32 (param i32)))
+  (import "symbolic" "assume_positive_i32" (func $positive_i32 (param i32)))
+  (import "symbolic" "assert" (func $assert_i32 (param i32)))
+
+  (func $start
+    (local $x i32)
+    (local.set $x (call $i32_symbol))
+    (call $assume_i32 (local.get $x))
+    (call $positive_i32 (local.get $x))
+    (call $assert_i32 (local.get $x))
```

Comment:
It would be nice to have different tests for each of them. Here using `positive_i32` and `assume_i32` on the same variable is redundant and doesn't guarantee proper testing of both.

Reply:
It should be better now
https://github.com/OCamlPro/owi/pull/96/commits/d877c3859fb00b78142516c0cbb397c53baec90f

Reply:
Well, I still believe it's not enough. Doing `positive_i32` and `assert_i32` on the same variable seems quite unlikely to happen in real code and it is still not clear what this is testing.

Reply:
May I offer a suggestion? Essentially `positive_i32` is just a more efficient way to write: 

```wast
(call $assume_i32 (i32.ge (i32.const 0) (local.get $x)))
``` 

We can easily encode such tests that prove integer overflow problems:
 
$x \geq 0  \wedge y > 1 \Rightarrow (x + y) > 0$

This inequality translates to the following WebAssembly program:

```wast
  (func $start
    (local $x i32)
    (local $y i32)
    (local.set $x (call $i32_symbol))
    (local.set $y (call $i32_symbol))
    (call $positive_i32 (local.get $x)) ;; x >= 0
    (call $assume_i32 (i32.gt_s (local.get $y) (i32.const 0)) ;; y > 0
    (if (i32.le_s (i32.add (local.get $x) (local.get $y)) (i32.const 0))
      (then unreachable))
    ;; Alternatively
    (call $assert_i32 (i32.gt_s (i32.add (local.get $x) (local.get $y)) (i32.const 0)))
)
```

Reply:
So your suggestion is to remove the function ? It thinks it's nice to have it as an illustration for some examples.

Reply:
Well, perhaps I sounded dismissive of `positive_i32`, but that was not my intention. I believe it is useful.

In my comment, I'm suggesting an alternative way to rewrite this test while maintaining its existing structure—specifically, using `positive_i32` once, `assume_i32` once, and `assert_i32` once. Instead of directly using `assert_i32` on a variable, I propose applying it to the expression $x + y > 0$. This allows the assert to quickly find the overflow bug, showing its correct behaviour.

Reply:
Oh okay my bad, I read to quickly. Indeed, I think the proposed program makes sense as a test. :-)

Reply:
Thanks for feedbacks @filipeom and @zapashcanon 
It was a bit difficult for me to write the right test ;)